src/qumin/representations/__init__.py

```diff
@@ -97,9 +98,8 @@ def get_unknown_segments(forms, unknowns, name):
         paradigms = paradigms.sample(sample)
 
     def aggregator(s):
-        if s.isnull().all():
-            return None
-        return ";".join(s.values)
+        tpl = tuple(s[s.apply(lambda x: form_dic[x] != '')])
```

Comment:
I think this is clearer as `tuple(form_id for form_id in s if form_dic[form_id] != '')`. We don't expect this to be big enough to require fancy apply, although we could check that there's no speed regression. If it needs to be fast and fancy, then `s.map(form_dic) != ''` is better than `s.apply(lambda x:form_dic[x] != '')`.

Also, let's call it `form_ids` rather than `tpl`, perhaps ? 

(Note: Let's keep in mind we went from a table of strings to a table of tuples)

Reply:
For ref: https://pandas.pydata.org/docs/reference/api/pandas.Series.map.html